test/relations.test.js

```diff
@@ -1726,6 +2322,24 @@ describe('relations', function () {
       });
     });
 
+    it('should allow to create belongsTo model in beforeCreate hook with promises', function (done) {
+      var mind;
+      Fear.beforeCreate = function (next) {
+        this.mind.create()
```

Comment:
`beforeCreate` hook is deprecated, plus I don't see any reason why we should suspect the promisification to work differently when invoked from a hook.

I am proposing to remove this particular unit test.

@raymondfeng thoughts?